common/smoke-tests/pom.xml

```diff
@@ -130,7 +130,7 @@
     <dependency>
       <groupId>com.azure</groupId>
       <artifactId>azure-security-keyvault-secrets</artifactId>
-      <version>4.3.0-beta.2</version> <!-- {x-version-update;com.azure:azure-security-keyvault-secrets;dependency} -->
+      <version>4.2.3</version> <!-- {x-version-update;com.azure:azure-security-keyvault-secrets;dependency} -->
```

Comment:
Why is KV secrets updated?

Reply:
Not too sure - that was there before I manually updated

Reply:
@vcolin7 and @g2vinay This seems like a valid change, but not sure why it should be happening in this PR. Could you confirm why we are seeing this change and if should be checked in?

Reply:
I notice, Yijun's PR added the x-version tags to smoke tests.
But, it didn't use dep version for KV, while it used dep x-version tag, so that looks incorrect to me in his PR.

https://github.com/Azure/azure-sdk-for-java/commit/4666053d1d893bd62f683d4d9173caf4e647769c#diff-f72ca982328bdb85aef7520207fbd08c38b15d828847b368b7d052172601c88f

 

I think it should be fine to keep this change in your PR. 

Since previously, smoke tests were using the stable release.

 

But, going forward, I'd debate using current tags on these, since smoke tests ideally should spot issues in the source code, I think.

Reply:
This change was initially introduced in https://github.com/Azure/azure-sdk-for-java/pull/18005/files and ideally should have been using 4.2.3. So this is good to check in for now.